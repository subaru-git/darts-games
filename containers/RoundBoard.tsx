import { Table, TableContainer, Tbody, Td, Tr } from "@chakra-ui/react";
import React, { FC } from "react";

type RoundBoardProps = {
  score: point[][];
};

const RoundBoard: FC<RoundBoardProps> = ({ score }) => {
  return (
    <>
      <TableContainer>
        <Table variant="simple">
          <Tbody>
            {score.map((round, i) => (
              <Tr key={`round-${i}`}>
                {round.map((point, j) => (
                  <Td key={`point-${i}-${j}`}>{point}</Td>
                ))}
              </Tr>
            ))}
          </Tbody>
        </Table>
      </TableContainer>
    </>
  );
};

export default RoundBoard;
