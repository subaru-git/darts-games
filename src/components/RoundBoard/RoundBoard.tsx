import React, { FC } from 'react';

type RoundBoardProps = {
  score: point[][];
};

const RoundBoard: FC<RoundBoardProps> = ({ score }) => {
  return (
    <div className='max-h-[300px] overflow-y-auto'>
      <table className='w-full table-fixed'>
        <tbody>
          {score.map((round, i) => (
            <tr key={`round-${i}`} className='border-b py-2'>
              {round.map((point, j) => (
                <td
                  key={`point-${i}-${j}`}
                  className='py-2 text-center font-semibold italic text-gray-500'
                >
                  {point === '0' ? '-' : point}
                </td>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default RoundBoard;
