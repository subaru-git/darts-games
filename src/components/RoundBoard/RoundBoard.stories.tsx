import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import RoundBoard from './RoundBoard';

export default {
  title: 'Components/RoundBoard',
  component: RoundBoard,
} as ComponentMeta<typeof RoundBoard>;

const Template: ComponentStory<typeof RoundBoard> = (args) => <RoundBoard {...args} />;

export const Default = Template.bind({});
Default.args = {
  score: [
    ['20T', '20', 'S-BULL'],
    ['19', '19T', '0'],
  ],
};

export const Overflow = Template.bind({});
Overflow.args = {
  score: [
    ['20T', '20', 'S-BULL'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
    ['19', '19T', '0'],
  ],
};
