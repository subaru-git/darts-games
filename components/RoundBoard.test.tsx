import { render, screen } from '@testing-library/react'
import RoundBoard from './RoundBoard'

test('should rendering', () => {
  const { container } = render(<RoundBoard score={[['20T', '19T', '18T']]} />)
  expect(screen.getAllByText('20T')).toHaveLength(1)
  expect(screen.getAllByText('19T')).toHaveLength(1)
  expect(screen.getAllByText('18T')).toHaveLength(1)
  expect(container).toMatchSnapshot()
})
