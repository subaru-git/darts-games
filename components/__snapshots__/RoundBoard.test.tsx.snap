// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="chakra-table__container css-0"
    >
      <table
        class="chakra-table css-0"
        role="table"
      >
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-0"
              role="gridcell"
            >
              20T
            </td>
            <td
              class="css-0"
              role="gridcell"
            >
              19T
            </td>
            <td
              class="css-0"
              role="gridcell"
            >
              18T
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
